src/main/java/com/frcteam3636/frc2024/subsystems/intake/IntakeIO.kt

```diff
@@ -1,48 +1,45 @@
 package com.frcteam3636.frc2024.subsystems.intake
 
 import com.frcteam3636.frc2024.CANDevice
+import com.frcteam3636.frc2024.Robot
 import com.revrobotics.CANSparkFlex
 import com.revrobotics.CANSparkLowLevel
 import com.revrobotics.CANSparkMax
-import com.revrobotics.SparkAbsoluteEncoder
 import edu.wpi.first.math.geometry.Rotation2d
-import edu.wpi.first.math.util.Units
-import org.littletonrobotics.junction.AutoLog
+import edu.wpi.first.math.system.plant.DCMotor
+import edu.wpi.first.wpilibj.simulation.FlywheelSim
+import org.littletonrobotics.junction.LogTable
+import org.littletonrobotics.junction.inputs.LoggableInputs
 
 interface IntakeIO {
-    @AutoLog
-    class IntakeInputs {
-        var armAngle = Rotation2d()
-        var armAngularVelocityHz = Rotation2d()
+    class IntakeInputs: LoggableInputs {
+        var overTheBumperFeedVelocityHz = Rotation2d()
+        var underTheBumperRollersVelocityHz = Rotation2d()
+
+        override fun toLog(table: LogTable?) {
+            table?.put("OTB Feed Velocity", overTheBumperFeedVelocityHz.radians)
+            table?.put("UTB Roller Velocity", underTheBumperRollersVelocityHz.radians)
+        }
+
+        override fun fromLog(table: LogTable?) {
+            table?.get("OTB Feed Velocity", 0.0)?.let { overTheBumperFeedVelocityHz = Rotation2d.fromRadians(it) }
+            table?.get("UTB Roller Velocity", 0.0)?.let { underTheBumperRollersVelocityHz = Rotation2d.fromRadians(it) }
+        }
     }
 
     fun updateInputs(inputs: IntakeInputs)
 
     fun setOverBumperFeed(speed: Double) {}
     fun setUnderBumperRoller(speed: Double) {}
-    fun moveArm(speed: Double) {}
 }
 
 class IntakeIOReal: IntakeIO {
-    companion object {
-        const val ARM_GEAR_RATIO = 1.0
-    }
-
-    private var armMotor = CANSparkMax(CANDevice.OverTheBumperIntakeArm.id, CANSparkLowLevel.MotorType.kBrushless)
-    private val armEncoder = armMotor.getAbsoluteEncoder(SparkAbsoluteEncoder.Type.kDutyCycle)
-
     private var overTheBumperFeed = CANSparkMax(CANDevice.OverTheBumperIntakeFeed.id, CANSparkLowLevel.MotorType.kBrushless)
     private var underTheBumperRoller = CANSparkFlex(CANDevice.UnderTheBumperIntakeRoller.id, CANSparkLowLevel.MotorType.kBrushless)
 
-    init {
-        armEncoder.velocityConversionFactor = Units.rotationsToRadians(1.0) * ARM_GEAR_RATIO / 60
-        armEncoder.positionConversionFactor = Units.rotationsToRadians(1.0) * ARM_GEAR_RATIO
-
-        armMotor.burnFlash()
-    }
     override fun updateInputs(inputs: IntakeIO.IntakeInputs) {
-        inputs.armAngle = Rotation2d(armEncoder.position)
-        inputs.armAngularVelocityHz = Rotation2d(armEncoder.velocity)
+        inputs.overTheBumperFeedVelocityHz = Rotation2d(overTheBumperFeed.encoder.velocity)
```

Comment:
Although it's more readable, we probably don't want to use property access syntax here since it allocates two new objects in each periodic call.

Reply:
`.encoder` doesn't create a new encoder except on the first access